File: samples/java/SupportChat/Server/Configuration/src/main/java/systems/zlink/samples/supportchat/server/configuration/SampleNames.java
package systems.zlink.samples.supportchat.server.configuration;

import systems.zlink.contracts.core.RoutingId;

public final class SampleNames {
    public static final String ApiChannel = "supportchat-api";
    public static final String SupportChannel = "supportchat-support";
    public static final String SupportActorMesh = "supportchat-actors";
    public static final RoutingId SupportNodeRoutingId = RoutingId.from("support");
    public static final String SupportActorType = "support-user";
    public static final String ConversationSpotType = "supportchat.conversation";
    public static final String StreamNode = "supportchat-session";
    public static final int AgentCapacity = 3;
    public static final String ConversationIdMetadataKey = "ConversationId";
    public static final String ServerEvidenceMarker = "supportchat-server-evidence=completed";
    public static final String ClientMarker = "supportchat=completed";

    private SampleNames() {}

    public static final class Roles {
        public static final String Customer = "Customer";
        public static final String Agent = "Agent";

        private Roles() {}
    }

    public static final class Statuses {
        public static final String WaitingForAgent = "WaitingForAgent";
        public static final String Active = "Active";
        public static final String WaitingForClose = "WaitingForClose";
        public static final String Closed = "Closed";

        private Statuses() {}
    }
}

File: samples/kotlin/SupportChat/Server/Configuration/src/main/kotlin/systems/zlink/samples/kotlin/supportchat/server/configuration/SampleNames.kt
package systems.zlink.samples.kotlin.supportchat.server.configuration

import systems.zlink.samples.kotlin.supportchat.shared.contracts.ChatMessageNotify
import systems.zlink.samples.kotlin.supportchat.shared.contracts.ConversationAssignedNotify
import systems.zlink.samples.kotlin.supportchat.shared.contracts.ConversationClosedNotify
import systems.zlink.samples.kotlin.supportchat.shared.contracts.ConversationIdleNotify
import systems.zlink.samples.kotlin.supportchat.shared.contracts.ParticipantJoinedNotify
import systems.zlink.samples.kotlin.supportchat.shared.contracts.TypingChangedNotify

object SampleNames {
    const val ApiChannel = "supportchat.api"
    const val SupportChannel = "supportchat.support"
    const val SupportActorType = "supportchat.user"
    const val ConversationSpotType = "supportchat.conversation"
    const val SupportSpotNode = "supportchat.support.node"
    const val SessionSpotNode = "supportchat.session.node"
    const val SupportEntrySpotNode = "supportchat.entry.node"
    const val SupportConversationSpotNode = "supportchat.conversation.node"
    const val SupportSpotDiscovery = "supportchat.support.spots"
    const val StreamNode = "supportchat.stream"
    const val AgentCapacity = 3
    const val ConversationIdMetadataKey = "ConversationId"

    const val ServerEvidenceMarker = "supportchat-server-evidence=completed"
    const val ClientMarker = "supportchat=completed"

    val ParticipantJoinedPacket: String = ParticipantJoinedNotify::class.java.simpleName
    val ConversationAssignedPacket: String = ConversationAssignedNotify::class.java.simpleName
    val ChatMessagePacket: String = ChatMessageNotify::class.java.simpleName
    val TypingChangedPacket: String = TypingChangedNotify::class.java.simpleName
    val ConversationIdlePacket: String = ConversationIdleNotify::class.java.simpleName
    val ConversationClosedPacket: String = ConversationClosedNotify::class.java.simpleName
}

object SupportChatRoles {
    const val Customer = "Customer"
    const val Agent = "Agent"
}

object ConversationStatuses {
    const val WaitingForAgent = "WaitingForAgent"
    const val Active = "Active"
    const val WaitingForClose = "WaitingForClose"
    const val Closed = "Closed"
}
